operator-common/src/main/java/io/strimzi/operator/common/featuregates/FeatureGates.java

```diff
@@ -43,13 +39,8 @@ public FeatureGates(String featureGateConfig) {
                 boolean value = '+' == featureGate.charAt(0);
                 featureGate = featureGate.substring(1);
 
-                switch (featureGate) {
-                    case CONTINUE_ON_MANUAL_RU_FAILURE:
-                        setValueOnlyOnce(continueOnManualRUFailure, value);
-                        break;
-                    default:
-                        throw new InvalidConfigurationException("Unknown feature gate " + featureGate + " found in the configuration");
-                }
+                // TODO: Once we have new FeatureGate, we should add back the switch, as it was here before
+                throw new InvalidConfigurationException("Unknown feature gate " + featureGate + " found in the configuration");
```

Comment:
Why not keep the switch but just leave a single default case? That way it's simpler to put new cases back in

Reply:
Yeah but checkstyle or spotbugs check will fail in that case. I had it in previous commits, but when I was changing the `hashCode()` and `equals()` it appeared as wrong. That's why I removed the switch, added the comment.

Reply:
I can add the switch back, just comment it out. That way the spotbugs check will not fail and it will be easier for the contributor to add it back.

Reply:
Added following to the code:

```
                // TODO: Once we have new FeatureGate, we should add back the switch, example is below
                // The switch is not implemented here to pass the spotbugs/checkstyle check.
                // switch (featureGate) {
                //    case CONTINUE_ON_MANUAL_RU_FAILURE:
                //        setValueOnlyOnce(continueOnManualRUFailure, value);
                //        break;
                //    default:
                //        throw new InvalidConfigurationException("Unknown feature gate " + featureGate + " found in the configuration");
                // }

                throw new InvalidConfigurationException("Unknown feature gate " + featureGate + " found in the configuration");
```